LayoutTests/inspector/formatting/resources/javascript-tests/for-statements-expected.js

```diff
@@ -8,6 +8,8 @@ for (; 1;)
     1
 for (;; 1)
     1
+for(;x?.y;)
```

Comment:
can you add some other tests like the following so we can see how this behaves in some slightly more complex scenarios?
```js
for (a;b?.c;)
for (a;b?.c;d)
for (;b?.c;d)

for (a?.b;;)
for (a?.b;c;)
for (a?.b;;d)
for (a?.b;c;d)

for (;;c?.d)
for (a;;c?.d)
for (;b;c?.d)
for (a;b;c?.d)

// etc.
```
since it seems like there's some oddities regarding optional chaining in `for` expressions it'd be nice to at least have some idea for how this behaves in as many scenarios as we can think of right now

that way, if we encounter anything unexpected we can either try to fix it or file some bugs on esprima (though perhaps our checkout is just old?)